filepathfilter/filepathfilter.go

```diff
@@ -143,6 +143,12 @@ func (w *wm) String() string {
 func NewPattern(p string) Pattern {
 	p = filepath.Clean(p)
 
+	// Special case: "*" and "*.*" should match everything to match existing
+	// behavior.
+	if p == "*" || p == "*.*" {
+		p = "**/*"
+	}
```

Comment:
I am confused here. Shouldn't `*` match everything also with the new behaviour too? I think `.gitattributes` handles it that way...

`*.*` would match all file/folder names with a dot in the new behaivior too, no?

Can you help me understand why we need this conversion?

Reply:
> I am confused here. Shouldn't `*` match everything also with the new behaviour too? I think `.gitattributes` handles it that way...

By default `*` is _not_ allowed to cross path separators, but with `WM_PATHNAME`, it can (behaving in-effect as `**`) [[1]](https://github.com/git/git/blob/v2.16.1/wildmatch.c#L110-L111). To the best of my knowledge, you are correct that this is the behavior that `.gitattributes` exhibits [[2]](https://github.com/git/git/blob/v2.16.1/dir.c#L1032-L1034).

I added support for this in https://github.com/git-lfs/wildmatch/pull/3, but removed it in https://github.com/git-lfs/wildmatch/commit/b6f7a2045f03a3b2f0cdfd677e809c95d2e5db6c when the complexity became too large to handle for this one particular exception.

Thus, without `WM_PATHNAME` support in `git-lfs/wildmatch`, we have to rewrite this pattern _as if_ `WM_PATHNAME` did exist in our implementation, which is `**/*` as above. In English, that's "match any number of leading directories, or none at all, and then any filename".

> `*.*` would match all file/folder names with a dot in the new behaivior too, no?

Correct, ~~but this was a special case of the prior implementation where instead of matching any filename with a dot in it, it would instead match any filename at all~~, and I misremembered that this was indeed a special case. I have reverted this behavior in 014fd601.